sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestHelper.scala

```diff
@@ -108,4 +126,23 @@ trait SQLQueryTestHelper {
         (emptySchema, Seq(e.getClass.getName, e.getMessage))
     }
   }
+
+  protected def splitWithSemicolon(seq: Seq[String]): Array[String] = {
+    seq.mkString("\n").split("(?<=[^\\\\]);")
+  }
+
+  protected def splitCommentsAndCodes(input: String): (Array[String], Array[String]) =
+    input.split("\n").partition { line =>
+      val newLine = line.trim
+      newLine.startsWith("--") && !newLine.startsWith("--QUERY-DELIMITER")
+    }
+
+  /** Returns all the files (not directories) in a directory, recursively. */
+  protected def listFilesRecursively(path: File): Seq[File] = {
```

Comment:
I assume these 3 method are just copied from the existing code

Reply:
Yes, I had moved these methods from their original place elsewhere in this file. I moved them back now to simplify the diff.